### PR TITLE
fix: Use SDK GetULimit

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: cmd/ulimit_unix.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func setUlimit(ulimit uint64) error {
 	return syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rLimit)
 }
 
-func GetUlimit() (syscall.Rlimit, error) {
+func getUlimit() (syscall.Rlimit, error) {
```
Comment (erezrokah) [inline cmd/ulimit_unix.go:47]: No need for this to be public, and I think we should use the SDKs `limit.GetUlimit()` but will leave that to another PR

Comment (erezrokah) [reply cmd/ulimit_unix.go]: Done in https://github.com/cloudquery/cloudquery/pull/1048/commits/4ad7eb7f87dd26a0fece3f27b5c24d7339c3611e

File: cmd/ulimit_unix_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+package cmd
+
+import (
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_setUlimit(t *testing.T) {
```
Comment (erezrokah) [inline cmd/ulimit_unix_test.go:12]: Very stupid test, probably can be improved by running on MacOS and also setting ulimit before and ensuring it was changed later

File: cmd/ulimit_unix_test.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_setUlimit_when_low(t *testing.T) {
+	// set to low values
+	err := syscall.Setrlimit(syscall.RLIMIT_NOFILE, &syscall.Rlimit{Cur: 10, Max: 20})
+	require.NoError(t, err)
+
+	err = setUlimit(ulimitUnix)
```
Comment (erezrokah) [inline cmd/ulimit_unix_test.go:17]: Setting the limit the second time fails

File: cmd/ulimit_unix.go [hunk 4]
```diff
@@ -41,11 +42,5 @@ func setUlimit(ulimit uint64) error {
...
 	}
 
-	return syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rLimit)
-}
-
-func GetUlimit() (syscall.Rlimit, error) {
-	var rLimit syscall.Rlimit
-	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
-	return rLimit, err
+	return syscall.Setrlimit(syscall.RLIMIT_NOFILE, &syscall.Rlimit{Cur: rLimit.Cur, Max: rLimit.Max})
```
Comment (roneli) [inline cmd/ulimit_unix.go:45]: maybe this should be moved to the SDK. Also removing the need to seperate ulimit files, and where we call it in root just do limit.Set etc' and in windows it will do nothing etc'.

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, suggested to maybe completely remove the low level ulimit calls here and move them 100% to SDK also reducing the ulimit_<os> bloat.

File: cmd/ulimit_unix.go (hunk 4 above)
Comment (erezrokah) [reply cmd/ulimit_unix.go:45]: Makes sense, I was able to make the test fail so looking into it

**Comment by erezrokah** [comment]
Per an internal discussion @shimonp21 will open a separate PR to handle setting the ulimit as we can only increase the hard limit when running in `root` permissions
